pkg/virt-handler/vm.go

```diff
@@ -1167,11 +1167,7 @@ func (c *VirtualMachineController) updatePausedConditions(vmi *v1.VirtualMachine
 	// Update paused condition in case VMI was paused / unpaused
 	if domain != nil && domain.Status.Status == api.Paused {
 		if !condManager.HasCondition(vmi, v1.VirtualMachineInstancePaused) {
-			reason := domain.Status.Reason
-			if c.isVMIPausedDuringMigration(vmi) {
-				reason = api.ReasonPausedMigration
-			}
-			calculatePausedCondition(vmi, reason)
+			calculatePausedCondition(vmi, domain.Status.Reason)
```

Comment:
How does this fix the issue? IIUC it only moves the logic carried out by the lines that were removed here into `calculatePausedCondition`? Can you add a test for it?

Reply:
I will add some unit test for it.

Reply:
@0xFelix unit test [added](https://github.com/kubevirt/kubevirt/compare/3d9797a3b3e9aca02e0e80cd3a789b3ccae84203..8a4d25de28575d5b7c0ee1005690522c30c0d9cd)